libmscore/measure.cpp

```diff
@@ -2802,6 +2802,9 @@ bool Measure::hasVoices(int staffIdx, Fraction stick, Fraction len) const
                                           break;
                                           }
                                     }
+                              // same if stem or hook(s) are visible
+                              if ((c->stem() || c->hook()) && c->visible())
+                                    v = true;
```

Comment:
Maybe moving this to above the `notes()` check and also add a `break;`, to handle the easy and most probable cases first?
Or would that be considered 'micro-optimization'?